src/WebJobs.Script.WebHost/Security/KeyManagement/KubernetesSecretsRepository.cs

```diff
@@ -69,6 +70,7 @@ public async Task WriteAsync(ScriptSecretsType type, string functionName, Script
                 throw new InvalidOperationException($"{nameof(KubernetesSecretsRepository)} is readonly when no {EnvironmentSettingNames.AzureWebJobsKubernetesSecretName} is specified.");
             }
 
+            functionName = functionName?.ToLowerInvariant();
```

Comment:
It depends on the `ScriptSecretsType`. If it's `ScriptSecretsType.Host` then it can be null or empty. I can add a validation, though other ISecretsRepository don't do it atm.

Reply:
done